src/main/java/org/opentripplanner/graph_builder/module/AddTransitModelEntitiesToGraph.java

```diff
@@ -124,29 +142,183 @@ private void addGroupsOfStationsToGraph(Graph graph) {
             Vertex fromVertex = stopNodes.get(pathway.getFromStop());
             Vertex toVertex = stopNodes.get(pathway.getToStop());
 
-            if(fromVertex != null && toVertex != null) {
-                if (pathway.isWheelchairTraversalTimeSet()) {
-                    new PathwayEdge(fromVertex, toVertex, pathway.getTraversalTime(), pathway.getWheelchairTraversalTime());
-                } else {
-                    new PathwayEdge(fromVertex, toVertex, pathway.getTraversalTime());
+            if (fromVertex != null && toVertex != null) {
+                // Elevator
+                if (pathway.getPathwayMode() == 5) {
+                    createElevatorEdgesAndAddThemToGraph(graph, pathway, fromVertex, toVertex);
+                }
+                else {
+                    new PathwayEdge(
+                        fromVertex,
+                        toVertex,
+                        pathway.getName(),
+                        pathway.getTraversalTime(),
+                        pathway.getLength(),
+                        pathway.getStairCount(),
+                        pathway.getSlope(),
+                        pathway.isPathwayModeWheelchairAccessible()
+                    );
+                    if (pathway.isBidirectional()) {
+                        new PathwayEdge(
+                            toVertex,
+                            fromVertex,
+                            pathway.getReversedName(),
+                            pathway.getTraversalTime(),
+                            pathway.getLength(),
+                            -1 * pathway.getStairCount(),
+                            -1 * pathway.getSlope(),
+                            pathway.isPathwayModeWheelchairAccessible()
+                        );
+                    }
                 }
             }
             else {
-                if(fromVertex == null) {
-                    LOG.warn("The 'fromVertex' is missing for pathway from stop: " + pathway.getFromStop().getId());
+                if (fromVertex == null) {
+                    LOG.warn("The 'fromVertex' is missing for pathway from stop: " + pathway
+                        .getFromStop()
+                        .getId());
                 }
-                if(toVertex == null) {
-                    LOG.warn("The 'toVertex' is missing for pathway to stop: " + pathway.getToStop().getId());
+                if (toVertex == null) {
+                    LOG.warn("The 'toVertex' is missing for pathway to stop: " + pathway
+                        .getToStop()
+                        .getId());
                 }
             }
         }
     }
 
+    /**
+     * Create elevator edges from GTFS pathways. As GTFS elevators are not vertices, but edges in
+     * the GTFS pathway model, we have to model each possible movement as an onboard-offboard pair,
+     * instead of having only one set of vertices per level and edges between them.
+     */
```

Comment:
Not specific to GTFS, it is not implemented in the NeTEx import yet, but hopefully if we add that this code will be reused without any modifications.